adaptors_task/student/include/impl.h

```diff
@@ -11,4 +12,62 @@
  *
  * @return true if expression is valid
  */
-bool isValid(const std::string& source);
+bool isValid(const std::string& source)
+{
+    std::stack<char> s;
+
+    for(const char& i: source) {
+        switch (i) {
+            case '[':
+                s.push(']');
```

Comment:
please check cpp file and use existing functionality

Reply:
updated